File: scripts/unzip.js
import { execSync } from 'child_process';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const projectRoot = path.join(__dirname, '..');
const zipPath = path.join(projectRoot, 'package-lock.zip');
const extractPath = projectRoot;

try {
  console.log('Unzipping package-lock.zip...');
  execSync(`unzip -o "${zipPath}" -d "${extractPath}"`, { stdio: 'inherit' });
  console.log('Unzipping complete!');
} catch (error) {
  console.error('Error unzipping file:', error.message);
  process.exit(1);
}

File: scripts/unzip.py
import zipfile
import os

zip_path = '/vercel/share/v0-project/package-lock.zip'
extract_path = '/vercel/share/v0-project'

try:
    print('Unzipping package-lock.zip...')
    with zipfile.ZipFile(zip_path, 'r') as zip_ref:
        zip_ref.extractall(extract_path)
    print('Unzipping complete!')
except Exception as error:
    print(f'Error unzipping file: {str(error)}')
    exit(1)
